test/models/px/test_dxwrapper.py

```diff
@@ -0,0 +1,159 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+
+import numpy as np
+import pytest
+import torch
+
+from earth2studio.data import Random, fetch_data
+from earth2studio.models.dx import (
+    CorrDiffTaiwan,
+    DerivedSurfacePressure,
+    PrecipitationAFNOv2,
+)
+from earth2studio.models.px import FCN3, DiagnosticWrapper
+from earth2studio.utils.coords import map_coords
+
+
+class PhooFCN3Model(torch.nn.Module):
+    def forward(self, x, t, normalized_data: bool = False, replace_state: bool = False):
+        return x
+
+
+class PhooAFNOPrecipV2(torch.nn.Module):
+    def forward(self, x):
+        return x[:, :1, :, :]
+
+
+class PhooCorrDiff(torch.nn.Module):
+    img_out_channels = 4
+    img_resolution = 448
+    sigma_min = 0
+    sigma_max = float("inf")
+
+    def forward(self, x, img_lr, class_labels=None, force_fp32=False, **model_kwargs):
+        return x[:, :4]
+
+    def round_sigma(self, sigma):
+        return torch.as_tensor(sigma)
+
+
+@pytest.mark.parametrize("device", ["cpu", "cuda:0"])
+def test_fcn3_precip(device):
+    # Spoof models
+    fcn3_model = PhooFCN3Model()
+    px_model = FCN3(fcn3_model)
+
+    precipafnov2_model = PhooAFNOPrecipV2()
+    center = torch.zeros(20, 1, 1)
+    scale = torch.ones(20, 1, 1)
+    landsea_mask = torch.zeros(1, 1, 720, 1440)
+    orography = torch.zeros(1, 1, 720, 1440)
+
+    precip_model = PrecipitationAFNOv2(
+        precipafnov2_model, landsea_mask, orography, center, scale
+    ).to(device)
+
+    px_out_coords = px_model.output_coords(px_model.input_coords())
+    sp_model = DerivedSurfacePressure(
+        p_levels=[50, 100, 150, 200, 250, 300, 400, 500, 600, 700, 850, 925, 1000],
+        surface_geopotential=torch.zeros(721, 1440),
+        surface_geopotential_coords=OrderedDict(
+            {"lat": px_out_coords["lat"], "lon": px_out_coords["lon"]}
+        ),
+    )
+
+    wrapped_model = DiagnosticWrapper(
+        px_model=px_model, dx_models=[sp_model, precip_model]
+    ).to(device=device)
+
+    dc = {k: wrapped_model.input_coords()[k] for k in ["lat", "lon"]}
+    data = Random(dc)
+
+    (x, coords) = fetch_data(
+        data,
+        [np.datetime64("2025-08-21T00:00:00")],
```

Comment:
Can we parametrize the input time to also include a test with more than one timestamp?